### ISSUE TITLE
The default behavior of separateHeaderAndContentWithNewLine is unwelcome 

### ISSUE THREAD

**Issue by hammerandtongs** [orgzly/orgzly-android#529]
- [x ] I have searched for existing issues that may be the same as or related to mine.

separateNotesWithNewLine and separateHeaderAndContentWithNewLine made me unhappy by adding newlines around all of the drawers to my existing org files.

I would think default to NEVER would be preferable, especially for people with existing org files.

It was also non-obvious where to look for the behavior in the preferences, I actually found them by grepping the source code (I thought it was a bug). Maybe the org format preferences should be in Notes and Notebooks?

**Comment by nevenz** [comment]
I agree it's kinda buried deep. Perhaps the whole "Notebook files" section that's now under Sync could be moved to the top level.

**Comment by pbx** [comment]
I found this issue when looking for a solution to this: a file I had edited in Orgzly (but mostly use in Org mode) gained newlines around the content of (every? many?) entries. This was not limited to drawers IIRC. 

Is that the issue described here?

(Secondarily -- where _is_ that preference? I'm not finding it in "Notes & Notebooks".)

**Comment by nevenz** [comment]
These options are under `Sync / Org file format`.

> I found this issue when looking for a solution to this: a file I had edited in Orgzly (but mostly use in Org mode) gained newlines around the content of (every? many?) entries

I don't think it could be called an issue, as it can be controlled with options.

**Comment by hammerandtongs** [comment]
I'm inclined to think of it as an issue because it is disruptive to people using orgmode outside of orgzly.

Even on Android I'm using emacs via termux to edit the same files.

It may not be immediately obvious to people not using git in a structured way. They don't see their 1000 line org file being mangled by the orgzly defaults.

I think any default behavior that deviates from the orgmode defaults should actually be treated like a bug.

Think of it like a kids game of telephone. Different programs touching the same file in subtly or significantly ways over time leads to a garbled message.

Align to the format defaults of orgmode and then deviate via options (especially options that are in good findable spots, which these are not currently in).

I realize its a hard problem once people start customizing via elisp or orgzly options but getting the defaults as a base seems like the right choice.

Alternatively a note detailing the options and locations on the README to "emacs orgmode users" would be useful.

Thanks again for Orgzly, very useful.

**Comment by nevenz** [comment]
> I think any default behavior that deviates from the orgmode defaults should actually be treated like a bug.

How do you define Org mode's default for this though?

For "Separate notes with an empty line" closest thing is `org-blank-before-new-entry` probably, but it's set to `auto` by default. This could be implemented - app could figure out user's preference when org file is first loaded and save it per notebook.

I don't think there's anything in Org mode for an empty line before content - it's a matter of hitting Enter in Emacs. Most Worg files use it for example (I've actually ran them all though Orgzly and the diff of files with app's current defaults is considerably smaller). I also prefer that the users who never saw org file before see a more spaced out version of the file. BTW, there's #190 too.

> Alternatively a note detailing the options and locations on the README to "emacs orgmode users" would be useful.

There's http://www.orgzly.com/help#d25f20c, it could be added there.

Anyway, is Settings / Notes & Notebooks / Org file format a better location then?

**Comment by A6GibKm** [comment]
> How do you define Org mode's default for this though?

org-mode's default behaviour is to not do anything, which is different to formatting the entire file according to a preference. The real issue here is that after adding a property to a single header, the entire document gets formatted and not just the header in which the property was added.

Thanks for pointing out there is a setting for this.

**Comment by GPPassos** [comment]
> Anyway, is Settings / Notes & Notebooks / Org file format a better location then?
Definitely!

> org-mode's default behaviour is to not do anything, which is different to formatting the entire file according to a preference. The real issue here is that after adding a property to a single header, the entire document gets formatted and not just the header in which the property was added.

100% agree with that. I always used orgzly to visualise, but not to edit (which I mainly do on my PC), and found out this issue.
The current optiions seem not to cut it as well: any possible value for "Separate notes with an empty line" will not respect previous decisions by the user.
It is quite surprising that e.g. inserting a new item causes every single entry in the document to be modified, and either many empty lines will be deleted or will be created. Why does Orgzly need to modify the full document this way?

In any case, I still really like the app, thank you for your work.
